src/renderer.ts

```diff
@@ -419,9 +419,40 @@ const render = (format: Format, imageMappings: ImageMappings, excludeStyles = fa
             const { html, css: styles, js } = element;
 
             const atom = h(InteractiveAtom, { html, styles, js, format });
+            const fenced = `
+                <html>
+                    <head>
+                        <meta charset="utf-8">
+                        <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+                        <script>
+                            const fonts = [].slice.apply(window.parent.document.styleSheets)
+                                .filter(sheet => sheet.ownerNode.attributes?.getNamedItem('data-emotion-css')?.value?.includes('PageStyles'))
+                                .map(sheet => sheet.ownerNode.textContent)
+                                .join(' ')
+                            var css = document.createElement('style');
+                            css.textContent = fonts;
+                            document.head.appendChild(css);
+                        </script>
+                        <style>
+                            ${styles}
+                        </style>
+                    </head>
+                    <body>
+                        ${html}
+                        <script>
+                            ${js}
+                            function resize() {
+                                window.frameElement.height = document.body.offsetHeight;
+                            }
+                            window.addEventListener('resize', resize);
+                            resize();
+                        </script>
+                    </body>
+                </html>
+            `;
 
             if (format.design !== Design.Interactive) {
-                return h('iframe', null, atom);
+                return h('iframe', { srcdoc: fenced });
             } else {
                 return atom;
             }
```

Comment:
Am I correct in thinking that the `atom` value is only generated for full-page interactives, and the `fenced` value is generated in all other cases? If so, would it make sense to make their generation part of this `if` condition so that we avoid doing additional work?

```ts
if (format.design === Design.Interactive) {
    return h(InteractiveAtom, { html, styles, js, format });
}

const fenced = ...;
return h('iframe', { srcdoc: fenced });
```